wurst/systems/spawns/ResourceSpawnsUtils.wurst

```diff
@@ -0,0 +1,201 @@
+package ResourceSpawnsUtils
+
+import Composer
+
+// Standard library imports:
+import ErrorHandling
+import HashMap
+import Interpolation
+import TerrainUtils
+
+// Local imports:
+import Game
+import GameTimer
+import GameConfig
+import GeometryUtils
+import Pool
+
+function getRandomAllowedPoint(Pool<rect> allowedRegions, region restrainedSpawnRegion) returns vec2
+    let spawnRect = allowedRegions.random()
+
+    var max = 1000 // Prevents infinite loop
+    while true and max >= 0

```

Comment:
Can't you use a for loop for this? And I meant a global to control the maximum number. Obviously you'll also need a local.

Reply:
what do you mean? while is a loop

ya i'm just not sure where to put that global, don't think it matters tbh since that should not happen and if it happens the spawn will just be broken for that one version